packages/synthetics-sdk-broken-links/src/broken_links.ts

```diff
@@ -82,31 +83,65 @@ export async function runBrokenLinks(
   let browser: Browser;
   try {
     const options = processOptions(inputOptions);
+    let timeLimitTimeout;
+    let timeLimitresolver = () => {};
+    const timeLimitPromise = new Promise<void>((resolve) => {
```

Comment:
I see this timeLimitPromise being redeclared 3 times, and since its kind of intense logic, I think you may want to move this to a shared place.

ie `timeLimitPromise = getTimeLimitPromise()`, then just use that throughout.  Maybe you need to pass the timeout length.

Reply:
My current structure is
GlobalTimeout (250 seconds)
- checkOriginTimeout (time left in global timeout minus 0.5 secs)
- checkLinksTimeout (time left in global timeout minus 0.5 secs)

I think this it is necessary to have different timeouts rather than using the same timeLimit timeout in all 3 Promise.race() because otherwise the global timeout could occurs momentarily before either of the other two and then we would get no information from that part of the execution.